src/main/java/org/fungover/thunder/ClientHandler.java

```diff
@@ -23,11 +25,21 @@ public void handleConnection(Socket clientSocket) {
             } else {
                 clientSocket.close();
             }
-            while (clients.contains(clientSocket)) {
-                if (packageReader.isCleanDisconnect(clientSocket)) {
+            InputStream inputStream = clientSocket.getInputStream();
+            OutputStream outputStream = clientSocket.getOutputStream();
+            byte[] buffer = new byte[1024];
+            int bytesRead;
+
+            while (!clientSocket.isClosed()) {
+                bytesRead = inputStream.read(buffer);
+                if (PackageReader.isDisconnectPackage(bytesRead, buffer)) {
+                    System.out.println("Received MQTT DISCONNECT message from client");
                     clientSocket.close();
-                    clients.remove(clientSocket);
-                    break;
+                } else if (PingHandler.isPingRequest(buffer, bytesRead)) {
+                    System.out.println("Received MQTT PINGREQ message from client");
+                    PingHandler.sendPingResponse(outputStream);
+                } else {
+                    // Handle other types of messages
```

Comment:
I guess this is because of a rollback?

Reply:
Yes thank you for the feedback, ClienthHandler tests are currently not working, we commented the tests out and initiated a new issue (#60 ) for resolution